### PR TITLE
Azure Open AI V1

### PR DESCRIPTION
**Pull request by BennuFire**

### CODE REVIEW

**Review by alfredorubin96 — APPROVED** [review]

File: src/extensions/query-translator/state/QueryTranslatorThunks.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export const modelClientInitializationThunk =
     if (modelProvider && settings) {
       // Getting the correct ModelClient object
       let tmpClient = getModelClientObject(modelProvider, settings);
-
+      console.log(tmpClient);
```
Comment (nielsdejong) [inline src/extensions/query-translator/state/QueryTranslatorThunks.ts:39]: to remove

File: src/extensions/query-translator/QueryTranslatorConfig.ts [hunk 2]
```diff
@@ -58,29 +59,33 @@ export const QUERY_TRANSLATOR_CONFIG: QueryTranslatorConfig = {
...
-    //       default: [],
-    //     }
-    //   }
-    // },
+    AzureOpenAI: {
+      clientName: 'AzureOpenAI',
+      clientClass: AzureOpenAiClient,
+      settings: {
+        endpoint: {
+          label: 'AzureOpenAI EndPoint',
```
Comment (nielsdejong) [inline src/extensions/query-translator/QueryTranslatorConfig.ts:67]: 'Azure OpenAI Endpoint'

File: src/extensions/query-translator/QueryTranslatorConfig.ts [hunk 3]
```diff
@@ -58,29 +59,33 @@ export const QUERY_TRANSLATOR_CONFIG: QueryTranslatorConfig = {
...
+          type: SELECTION_TYPES.TEXT,
+          default: '',
+          hasAuthButton: false,
+          authentication: true,
+        },
+        apiKey: {
+          label: 'AzureOpenAI API Key / Subscription Key',
```
Comment (nielsdejong) [inline src/extensions/query-translator/QueryTranslatorConfig.ts:74]: Maybe just 'Subscription Key'?

**Review by nielsdejong — CHANGES_REQUESTED** [review]
Some suggestions

File: src/extensions/query-translator/state/QueryTranslatorThunks.ts (hunk 1 above)
Comment (BennuFire) [reply src/extensions/query-translator/state/QueryTranslatorThunks.ts:39]: Done

File: src/extensions/query-translator/QueryTranslatorConfig.ts (hunk 3 above)
Comment (BennuFire) [reply src/extensions/query-translator/QueryTranslatorConfig.ts:74]: Done

**Review by nielsdejong — APPROVED** [review]

**Pull request merged**